tests/unit/steps/test_steps_plan.py

```diff
@@ -743,19 +742,93 @@ def test_generate_common_plan(cli_backup, cli_args, model):
             PreUpgradeStep(
                 description="Backup mysql databases",
                 parallel=False,
                 coro=backup(model),
             )
         )
 
-    assert common_plan == expected_plan
+    pre_upgrade_steps = cou_plan._get_pre_upgrade_steps(mock_analysis_result, cli_args)
+
+    assert pre_upgrade_steps == expected_steps
+
+
+@pytest.mark.parametrize("upgrade_group", [CONTROL_PLANE, HYPERVISORS])
+@patch("cou.steps.plan._get_ceph_mon_post_upgrade_steps")
+def test_get_post_upgrade_steps_empty(mock_get_ceph_mon_post_upgrade_steps, upgrade_group):
+    """Test get post upgrade steps not run for control-plane or hypervisors group."""
+    args = MagicMock(spec_set=CLIargs)()
+    args.upgrade_group = upgrade_group
+
+    pre_upgrade_steps = cou_plan._get_post_upgrade_steps(MagicMock(), args)
+
+    assert pre_upgrade_steps == []
+    mock_get_ceph_mon_post_upgrade_steps.assert_not_called()
+
+
+@pytest.mark.parametrize("upgrade_group", [DATA_PLANE, None])
+@patch("cou.steps.plan._get_ceph_mon_post_upgrade_steps")
+def test_get_post_upgrade_steps_ceph_mon(mock_get_ceph_mon_post_upgrade_steps, upgrade_group):
+    """Test get post upgrad steps including ceph-mon."""
+    args = MagicMock(spec_set=CLIargs)()
+    args.upgrade_group = upgrade_group
+    analysis_result = MagicMock(spec_set=Analysis)()
+    mock_get_ceph_mon_post_upgrade_steps.return_value = [MagicMock()]
+
+    pre_upgrade_steps = cou_plan._get_post_upgrade_steps(analysis_result, args)
+
+    assert pre_upgrade_steps == mock_get_ceph_mon_post_upgrade_steps.return_value
+    mock_get_ceph_mon_post_upgrade_steps.assert_called_with(analysis_result.apps_data_plane)
+
+
+def test_get_ceph_mon_post_upgrade_steps_zero(model):
+    """Test get post upgrade step for ceph-mon without any ceph-mon app."""
+    analysis_result = MagicMock(spec_set=Analysis)()
+    analysis_result.apps_control_plane = []
+
+    step = cou_plan._get_ceph_mon_post_upgrade_steps(analysis_result)
+
+    assert bool(step) is False
+
+
+def test_get_ceph_mon_post_upgrade_steps_multiple(model):
+    """Test get post upgrade step for ceph-mon with multiple ceph-mon."""
+    machines = {"0": MagicMock(spec_set=COUMachine)}
+    units = {
+        "ceph-mon/0": COUUnit(
+            name="ceph-mon/0",
+            workload_version="17.0.1",
+            machine=machines["0"],
+        )
+    }
+    ceph_mon = CephMon(
+        name="ceph-mon",
+        can_upgrade_to="",
+        charm="ceph-mon",
+        channel="quincy/stable",
+        config={"source": {"value": "distro"}},
+        machines={},
+        model=model,
+        origin="ch",
+        series="focal",
+        subordinate_to=[],
+        units=units,
+        workload_version="17.0.1",
+    )
+
+    exp_steps = 2 * [
+        PostUpgradeStep(
+            "Ensure the 'require-osd-release' option matches the 'ceph-osd' version",
+            coro=app_utils.set_require_osd_release_option("cpeh-mon/0", model),
```

Comment:
I'm confused why the unit test didn't fail with this typo. Maybe we have a bug on `compare_step_coroutines`
```suggestion
            coro=app_utils.set_require_osd_release_option("ceph-mon/0", model),
```